subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/JdkJavaCompiler.java

```diff
@@ -63,7 +78,47 @@ public WorkResult execute(JavaCompileSpec spec) {
         if (JavaVersion.current().isJava9Compatible() && emptySourcepathIn(options)) {
             fileManager = (StandardJavaFileManager) SourcepathIgnoringProxy.proxy(standardFileManager, StandardJavaFileManager.class);
         }
-        return compiler.getTask(null, fileManager, null, options, null, compilationUnits);
+        JavaCompiler.CompilationTask task = compiler.getTask(null, fileManager, null, options, null, compilationUnits);
+
+        List<AnnotationProcessorDeclaration> annotationProcessors = spec.getEffectiveAnnotationProcessors();
+        if (annotationProcessors != null) {
+            task.setProcessors(instantiateProcessors(annotationProcessors, spec.getAnnotationProcessorPath()));
+        }
+        return task;
+    }
+
+    private List<Processor> instantiateProcessors(List<AnnotationProcessorDeclaration> declaredProcessors, List<File> annotationProcessorPath) {
+        if (declaredProcessors == null) {
+            return Collections.emptyList();
+        }
+        processorClassloader = new URLClassLoader(getUrls(annotationProcessorPath));
+        List<Processor> processors = new ArrayList<Processor>(declaredProcessors.size());
+        for (AnnotationProcessorDeclaration declaredProcessor : declaredProcessors) {
+            try {
+                Class<?> processorClass = processorClassloader.loadClass(declaredProcessor.getClassName());
+                Object processor = processorClass.newInstance();
+                processors.add((Processor) processor);
+            } catch (Exception e) {
+                throw new IllegalArgumentException(e);
+            }
+        }
+        return processors;
+    }
+
+    private URL[] getUrls(List<File> classpath) {
+        URL[] urls = new URL[classpath.size()];
+        for (int i = 0; i < urls.length; i++) {
+            try {
+                urls[i] = classpath.get(i).toURI().toURL();
+            } catch (MalformedURLException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        return urls;
+    }
+
+    private void cleanup() {
+        CompositeStoppable.stoppable(processorClassloader).stop();
```

Comment:
⭕️ Should you set it to `null` too?

Reply:
This whole object is GCd shortly after, so I don't think so.

Reply:
Ah sorry, this was in the old design anyway.